File: LEETCODE1600/Output/LEETCODE_Data/206-reverse-linked-list反转链表/pair_28/6.c
struct ListNode *Reverse(struct ListNode *head)
{
    if (head->next == NULL)
        return head;
    Reverse(head->next)->next = head;
    return head;
}

struct ListNode *reverseList(struct ListNode *head)
{
    if (head == NULL)
        return NULL;
    struct ListNode *end;
    struct ListNode *Copy_Head = head;
    while (Copy_Head->next != NULL)
        Copy_Head = Copy_Head->next;
    end = Copy_Head;
    Reverse(head)->next = NULL;
    return end;
}

File: LEETCODE1600/Output/intermediate/LEETCODE_AST/206-reverse-linked-list反转链表/rust_code/6.rs
impl Solution { 
    pub fn reverse_list(head: Option<Box<ListNode>>) -> Option<Box<ListNode>> { 
        let mut node = head; 
        let mut res = None;
        while let Some(t) = node { 
            res = Some(Box::new(ListNode{val: t.val, next: res}));  
            node = t.next; 
        }
        res 
    } 
}